lib/src/deprecations.ts

```diff
@@ -25,53 +25,103 @@ export function getDeprecationIds(
   });
 }
 
+/**
+ * Map between active compilations and the deprecation options they use.
+ *
+ * This is used to determine which options to use when handling host-side
+ * deprecation warnings that aren't explicitly tied to a particular compilation.
+ */
+export const activeDeprecationOptions: Map<Object, DeprecationOptions> =
+  new Map();
+
 /**
  * Shorthand for the subset of options related to deprecations.
  */
-export type DeprecationOptions = {
-  fatalDeprecations?: (DeprecationOrId | Version)[];
-  futureDeprecations?: DeprecationOrId[];
-  silenceDeprecations?: DeprecationOrId[];
-};
+export type DeprecationOptions = Pick<
+  Options<'sync'>,
+  'fatalDeprecations' | 'futureDeprecations' | 'silenceDeprecations'
+>;
 
 /**
  * Handles a host-side deprecation warning, either emitting a warning, throwing
- * and error, or doing nothing depending on the deprecation options used.
+ * an error, or doing nothing depending on the deprecation options used.
+ *
+ * If no specific deprecation options are passed here, then options will be
+ * determined based on the options of the active compilations.
  */
 export function warnForHostSideDeprecation(
   message: string,
   deprecation: Deprecation,
   options?: DeprecationOptions
 ): void {
+  if (options) throw Error(message);
```

Comment:
I don't understand this line. Is it left over from debugging?

Reply:
Yes, though I'd removed it before requesting review, so I think this was just GitHub's review interface being bad